homeassistant/components/bluesound/config_flow.py

```diff
@@ -0,0 +1,125 @@
+"""Config flow for bluesound."""
+
+import logging
+from typing import Any
+
+from pyblu import Player, SyncStatus
+import voluptuous as vol
+
+from homeassistant.components import zeroconf
+from homeassistant.config_entries import ConfigFlow, ConfigFlowResult
+from homeassistant.const import CONF_HOST, CONF_PORT
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+
+from .const import DOMAIN
+from .media_player import DEFAULT_PORT
+from .utils import format_unique_id
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class BluesoundConfigFlow(ConfigFlow, domain=DOMAIN):
+    """Bluesound config flow."""
+
+    VERSION = 1
+    MINOR_VERSION = 1
+
+    def __init__(self) -> None:
+        """Initialize the config flow."""
+        self._host: str | None = None
+        self._sync_status: SyncStatus | None = None
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Handle a flow initiated by the user."""
+        if user_input is not None:
+            session = async_get_clientsession(self.hass)
+            async with Player(
+                user_input[CONF_HOST], user_input[CONF_PORT], session=session
+            ) as player:
+                try:
+                    sync_status = await player.sync_status(timeout=1)
+                except TimeoutError:
+                    return self.async_abort(reason="cannot_connect")
+
+            await self.async_set_unique_id(
+                format_unique_id(sync_status.mac, user_input[CONF_PORT])
```

Comment:
Why is the port of significance? Isn't the Mac unique enough?

Reply:
There are apparently some devices which have multiple virtual media players running on different ports.

From the docs:

> The default port number is 11000. Players such as the NAD CI580, which has four players with one IP, use multiple ports.

Reply:
Uuugh, why did they invent this

Reply:
This seems to be a hifi audio thing :man_shrugging: Not the first brand I encountered, which had questionable API design :laughing: